client_internals_it_test.go

```diff
@@ -149,3 +152,53 @@ func (h *riggedInvocationHandler) Invoke(inv invocation.Invocation) (int64, erro
 	}
 	return h.Handler.Invoke(inv)
 }
+
+func TestClusterID(t *testing.T) {
+	it.SkipIf(t, "oss")
+	clientTester(t, func(t *testing.T, smart bool) {
+		ctx := context.Background()
+		cls1 := it.StartNewClusterWithOptions("clusterId-test-cluster1", 15701, it.MemberCount())
+		cls2 := it.StartNewClusterWithOptions("clusterId-test-cluster2", 16701, it.MemberCount())
+		defer cls2.Shutdown()
+		var wg sync.WaitGroup
+		wg.Add(1)
+		config1 := cls1.DefaultConfig()
+		config1.Cluster.ConnectionStrategy.Timeout = types.Duration(5 * time.Second)
+		config2 := cls2.DefaultConfig()
+		config := hz.Config{
+			Failover: cluster.FailoverConfig{
+				Enabled:  true,
+				TryCount: 0,
+			},
+		}
+		config.Failover.SetConfigs(config1.Cluster, config2.Cluster)
+		config.AddLifecycleListener(func(event hz.LifecycleStateChanged) {
+			if event.State == hz.LifecycleStateChangedCluster {
+				wg.Done()
+			}
+		})
+		c := it.MustClient(hz.StartNewClientWithConfig(ctx, config))
+		defer func(ctx context.Context, c *hz.Client) {
+			err := c.Shutdown(ctx)
+			if err != nil {
+				t.Error("client should had shutdown properly")
+			}
+		}(ctx, c)
+		ci := hz.NewClientInternal(c)
+		prevClusterId := ci.ClusterID()
```

Comment:
Do you think we should assert this against types.UUID{}?

Reply:
I guess, in this test scenario we made sure that `prevClusterId` is not nil or empty because of the statement, 
`c := it.MustClient(hz.StartNewClientWithConfig(ctx, config))`. If the client is started with no error, It is obvious that there is a ClusterID.